### DISCUSSION TITLE
Allow mdx extension in gatsby-theme-minimal-blog-core

### DISCUSSION THREAD

**Discussion by jsg2021** [LekoArts/gatsby-themes#1154] [Questions]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@lekoarts/gatsby-theme-minimal-blog-core@6.1.0` for the project I'm working on.

What would you think of allowing the `mdx` config option to allow additional plugins? I tried to turn mdx off from the theme, and copy that block and add my own extensions, but gatsby failed to start (maybe I'm missing something?)... extending your package with this line, simply allowed me to add the plugin.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@lekoarts/gatsby-theme-minimal-blog-core/gatsby-config.mjs b/node_modules/@lekoarts/gatsby-theme-minimal-blog-core/gatsby-config.mjs
index e514cc2..da19748 100755
--- a/node_modules/@lekoarts/gatsby-theme-minimal-blog-core/gatsby-config.mjs
+++ b/node_modules/@lekoarts/gatsby-theme-minimal-blog-core/gatsby-config.mjs
@@ -37,6 +37,7 @@ const config = (themeOptions) => {
           },
           extensions: [`.mdx`, `.md`],
           gatsbyRemarkPlugins: [
+			...(options.mdx.remarkPlugins || []),
             {
               resolve: `gatsby-remark-images`,
               options: {
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by LekoArts** [comment]
Hi!

With the `mdx` option you can completely disable the inclusion of the plugin in the core theme. You then can configure gatsby-plugin-mdx on your own. Thus there is no need for any changes. Can you share your code of what you tried?

**Comment by jsg2021** [comment]
I was attempting to avoid re-declaring all the mdx config just to add another plugin
```js
resolve: `@lekoarts/gatsby-theme-minimal-blog`,
      // See the theme's README for all available options
      options: {
        mdx: {
          remarkPlugins: [
            {
              resolve: 'gatsby-remark-embedder',
              options: {
                customTransformers: [vimeo],
              },
            },
          ],
        },
...
```

However, in hindsight, that may be the safest approach.

**Reply by jsg2021** [reply]
I just tried this change and I'm now getting graphql errors... looks like with `mdx:false` the benefit of using your theme is significantly reduced. :( (since the vast majority of page/data setup is omitted) 

```diff
diff --git a/gatsby-config.ts b/gatsby-config.ts
index b06a892..f537e0c 100755
--- a/gatsby-config.ts
+++ b/gatsby-config.ts
@@ -1,4 +1,6 @@
 import type { GatsbyConfig, PluginRef } from 'gatsby';
+import remarkGfm from 'remark-gfm';
+import rehypeMetaAsAttributes from '@lekoarts/rehype-meta-as-attributes';
 import 'dotenv/config';
 
 import vimeo from './src/utils/vimeo';
@@ -30,16 +32,7 @@ const config: GatsbyConfig = {
 			resolve: `@lekoarts/gatsby-theme-minimal-blog`,
 			// See the theme's README for all available options
 			options: {
-				mdx: {
-					remarkPlugins: [
-						{
-							resolve: 'gatsby-remark-embedder',
-							options: {
-								customTransformers: [vimeo],
-							},
-						},
-					],
-				},
+				mdx: false,
 				formatString: 'MMMM D, YYYY',
 				navigation: [
 					{
@@ -85,6 +78,35 @@ const config: GatsbyConfig = {
 				],
 			},
 		},
+		{
+			resolve: `gatsby-plugin-mdx`,
+			options: {
+				mdxOptions: {
+					remarkPlugins: [
+						remarkGfm,
+						{
+							resolve: 'gatsby-remark-embedder',
+							options: {
+								customTransformers: [vimeo],
+							},
+						},
+					],
+					rehypePlugins: [rehypeMetaAsAttributes],
+				},
+				extensions: [`.mdx`, `.md`],
+				gatsbyRemarkPlugins: [
+					{
+						resolve: `gatsby-remark-images`,
+						options: {
+							maxWidth: 960,
+							quality: 90,
+							linkImagesToOriginal: false,
+							backgroundColor: `transparent`,
+						},
+					},
+				],
+			},
+		},
 		{
 			resolve: 'gatsby-plugin-google-analytics',
 			options: {
```

```
ERROR #85923  GRAPHQL.VALIDATION

There was an error in your GraphQL query:

Cannot query field "post" on type "Query".

If you don't expect "post" to exist on the type "Query" it is most likely a typo. However, if you expect "post" to exist there are a couple
 of solutions to common problems:

- If you added a new data source and/or changed something inside gatsby-node/gatsby-config, please try a restart of your development
server.
- You want to optionally use your field "post" and right now it is not used anywhere.

It is recommended to explicitly type your GraphQL schema if you want to use optional fields.

File: src/@lekoarts/gatsby-theme-minimal-blog-core/templates/post-query.tsx:10:3

See our docs page for more info on this error: https://gatsby.dev/creating-type-definitions
```



... I am really hoping to avoid re-declaring all this config

**Reply by LekoArts** [reply]
`gatsby-remark-embedder` has to go into `gatsbyRemarkPlugins`, not`mdxOptions.remarkPlugins`.

Other than that, this should work. You need to set `mdx: false` in the `-core` theme and then redeclare the complete MDX plugin config + your additions.

**Reply by jsg2021** [reply]
by re-declare, do you also mean all the `gatsby-node` page definitions too? (which is what I'm trying to avoid) or _just_ the plugin config in `gatsby-config`?

**Reply by LekoArts** [reply]
> just the plugin config in gatsby-config

this

**Reply by jsg2021** [reply]
Sorry for the delayed response, just got a moment to try this... i couldn't get it to work.

I tried a new blank project with only the `mdx: false` config and copied the themes' mdx config into the project...

https://github.com/jsg2021/theme-options-mdx-example

When I start it, I get the same errors.

**Reply by melloware** [reply]
I cannot get this to work either. Same GraphQL errors
